src/server/webhookServer.ts

```diff
@@ -0,0 +1,95 @@
+import * as http from 'http';
+import debug from '../debug';
+import { IApiRequest } from '../api/request';
+import { IApiResponse } from '../api/response';
+
+type handleAliceRequestType = (request: IApiRequest) => Promise<IApiResponse>;
+
+export interface IWebhookServerConfig {
+  port: number;
+  webhookUrl: string;
+  options: object;
+  handleRequest: handleAliceRequestType;
+}
+export interface IWebhookServer {
+  start(): void;
+  stop(): void;
+}
+
+export class WebhookServer {
+  private server: http.Server;
+  private port: number;
+  private webhookUrl: string;
+  private _isStarted: boolean;
+  private _handleAliceRequest: handleAliceRequestType;
+
+  constructor(config: IWebhookServerConfig) {
+    this.port = config.port;
+    this.webhookUrl = config.webhookUrl;
+    this._handleAliceRequest = config.handleRequest;
+    this._isStarted = false;
+
+    debug(`server is listening: ${this.port}, ${this.webhookUrl}`);
+    this.server = http.createServer((request, response) =>
+      this._handleRequest(request, response),
+    );
+  }
+
+  private async _handleRequest(
+    request: http.IncomingMessage,
+    response: http.ServerResponse,
+  ): Promise<void> {
+    if (request.method !== 'POST' || request.url !== this.webhookUrl) {
+      response.statusCode = 400;
+      return response.end();
+    }
+
+    const requestBody = await WebhookServer._readRequest(request);
+    const responseBody = await this._handleAliceRequest(requestBody);
+    WebhookServer._sendResponse(response, responseBody);
+  }
+
+  private static _readRequest(
+    request: http.IncomingMessage,
+  ): Promise<IApiRequest> {
+    return new Promise((resolve, reject) => {
+      const body: Array<string | Buffer> = [];
+      request
+        .on('data', chunk => {
+          body.push(chunk);
+        })
+        .on('end', async () => {
+          const requestData = Buffer.from(body).toString();
+          resolve(JSON.parse(requestData));
+        })
+        .on('error', reject);
```

Comment:
There is no 'error' event on request. :-(
Sorry. You can leave it as is i'll fix it later.